lib/iris/fileformats/_nc_load_rules/helpers.py

```diff
@@ -356,10 +357,15 @@ def parse_cell_methods(nc_cell_methods):
             method_words = method.split()
             if method_words[0].lower() not in _CM_KNOWN_METHODS:
                 msg = "NetCDF variable contains unknown cell method {!r}"
-                warnings.warn(
-                    msg.format("{}".format(method_words[0])),
-                    category=UnknownCellMethodWarning,
-                )
+                msg = msg.format(method_words[0])
+                if cf_name:
+                    name = "{}".format(cf_name)
+                    msg = msg.replace("variable", "variable {!r}".format(name))
+                else:
+                    warnings.warn(
+                        msg,
+                        category=UnknownCellMethodWarning,
+                    )
```

Comment:
I think it might actually be worth adding a `msg = None` here to ensure this warning isn't repeated at the end of the function.